hack/codecov.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+CI_SERVER_URL=https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test
+COVER_PROFILE=${COVER_PROFILE:-coverage.out}
+JOB_TYPE=${JOB_TYPE:-"local"}
+
+# Default concurrency to four threads. By default it's the number of procs,
+# which seems to be 16 in the CI env. Some consumers' coverage jobs were
+# regularly getting OOM-killed; so do this rather than boost the pod resources
+# unreasonably.
+COV_THREAD_COUNT=${COV_THREAD_COUNT:-4}
+make -C "${REPO_ROOT}" go-test TESTOPTS="-coverprofile=${COVER_PROFILE}.tmp -covermode=atomic -coverpkg=./... -p ${COV_THREAD_COUNT}"
```

Comment:
The make target referred to in this script is `go-test` while ADO doesn't have that specific target.
Let's modify this script to include `test-unit` make target of ADO.
This will need further modifications to the `test-unit` target; instead of invoking the mage `test:unit` call let's specify the go test command in the target:
```
	CGO_ENABLED=1 go test $(TESTOPTS) ./internal/... ./cmd/... ./pkg/...
```
Also, you will also need to add the `TESTOPTS` variable to the Makefile:
```
TESTOPTS ?=-cover -race -v
```

Reply:
Will changing the test-unit command in this way break anything else?

Reply:
This is just modifying the test options by adding the `cover` flag so this won't break anything.

For affirmation, the CI tests just passed which runs against the latest changes made in the pull requests, hence no breaking changes.